core/raftlease/fsm.go

```diff
@@ -267,20 +267,20 @@ func (f *FSM) filteredLeases(getLocalTime func() time.Time, keys []lease.Key) ma
 			results[key] = f.infoFromEntry(localTime, key, entry)
 		}
 	}
-	f.mu.Unlock()
+	f.mu.RUnlock()
 	return results
 }
 
 func (f *FSM) allLeases(getLocalTime func() time.Time) map[lease.Key]lease.Info {
 	results := make(map[lease.Key]lease.Info)
-	f.mu.Lock()
+	f.mu.RLock()
 	localTime := getLocalTime()
 	for _, entries := range f.groups {
 		for key, entry := range entries {
 			results[key] = f.infoFromEntry(localTime, key, entry)
 		}
 	}
-	f.mu.Unlock()
+	f.mu.RUnlock()
```

Comment:
`RUnlock`should be invoked using `defer` to maintain consistency with the other pieces of code.

Reply:
defer was slower at one point, which was probably why it was written like this. In reality that's gone now, so probably worth refactoring for readability.

Reply:
I add the Go notice about the defer improvement for the sake of completeness https://golang.org/doc/go1.14#runtime